### ISSUE TITLE
Demo and Toxin GLA Ambush are very strong

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#1184]
Demo and Toxin GLA Ambush are very strong in comparison to regular Ambushes. Does it need change?

Stubbjax
> The reason Rebel Ambush + Demolitions is typically banned in rule-based games is because it is absolutely outrageous that $10,000+ of stuff can be destroyed in the blink of an eye, with almost no chance for counterplay. Destructive powers / abilities should always provide the opponent with some opportunity for anticipation / avoidance / denial, and otherwise feel incredibly frustrating and unfair to play against. The specialised ambush variants unfortunately ignore this rule, and are very poorly considered / designed. (A far better design would be if the Suicide ability had an initial cooldown like the capture upgrade, so that opponents had some time to react. The Toxin variant could just spawn fewer Rebels.)

### Proposal 1

Add short cooldown to Demo suicide ability.

### Proposal 2

Spawn fewer Toxin Rebels. 3/6/12 Rebels (from 4/8/16)

### Proposal 3

Make Toxin Rebels unable to shoot for a few seconds.

### Proposal 4

Spawn fewer Demo Rebels.  2/4/8 Rebels (from 4/6/10)

**Comment by ImTimK** [comment]
Pretty much some topic and some proposals :P 

https://github.com/TheSuperHackers/GeneralsGamePatch/issues/872

**Comment by ExiLe-ZH** [comment]
Pretty much same topic and same proposals :P 

https://github.com/TheSuperHackers/GeneralsGamePatch/issues/872

**Comment by KrewGR** [comment]
Someone will wrongfully cry about this change, but it'll be music to my ears.

**Comment by xezon** [comment]
> Pretty much same topic and same proposals :P

Dang. Let us close it again then.

**Comment by ExiLe-ZH** [comment]
I feel we haven't even discussed these things thoroughly

**Comment by xezon** [comment]
Discuss.

**Comment by ExiLe-ZH** [comment]
I'd like to test proposal 1 and experiment how much delay would be fair.

I definitely agree with proposal 2, make the numbers similar to Demo Rebel Ambush.

**Comment by Stubbjax** [comment]
A three-second delay would probably be fair.

But it may be a lot of work to achieve this because the suicide ability is implemented as a tertiary weapon and likely needs to be changed to a special ability for the initial cooldown to function.

**Comment by MTKing4** [comment]
There was another proposal comparable to the detonation delay for tox ambush

Proposal 3: make tox rebels unable to shoot for a few seconds

**Comment by MTKing4** [comment]
Wait is the tox ambush spawn more rebels than other ambushes?

**Comment by ExiLe-ZH** [comment]
> Wait is the tox ambush spawn more rebels than other ambushes?

GLA, Tox, Stealth: 4/8/16
Demo: 4/6/10

Originally Demo had same numbers as the other GLA's in earlier patches but EA realized it was OP with the Demo Ability.

Imo it's fair to give Tox the same treatment as the effectiveness vs armies feels overpowered.

**Comment by commy2** [comment]
### Proposal 4

- keep GLA & Stealth at 4/8/16 Rebels
- nerf Tox to 3/6/12 Tox Rebels (from 4/8/16)
- nerf Demo to 2/4/8 Rebels (from 4/6/10)

**Comment by ExiLe-ZH** [comment]
> ### Proposal 4
> 
> 
> 
> - keep GLA & Stealth at 4/8/16 Rebels
> 
> - nerf Tox to 3/6/12 Tox Rebels (from 4/8/16)
> 
> - nerf Demo to 2/4/8 Rebels (from 4/6/10)

Demo level 1 and 2 do sound a little sad now, what about 3/5/8?

**Comment by xezon** [comment]
I expect that reducing Demo Rebel counts will not weaken it considerably against blobs of vehicles, because a single Rebel deals a great amount of damage on its own. The likely biggest impact it would have is decreasing its total damage output versus structures to a point where it perhaps no longer is able to take out high armor structures. This in turn will shift the usefulness more towards use against vehicle blobs, which exacerbates the problem that Stubbjax has initially raised.

**Comment by ExiLe-ZH** [comment]
It's not just the damage though, yes vs vehicles higher rebel numbers is indeed overkill in terms of raw power, however higher numbers also increases the radius, because they spawn more spread out. 

That said, my suggestion of 3/5/8 is probably not a significant nerf vs vehicle blobs. 

Another suggestion in a topic https://github.com/TheSuperHackers/GeneralsGamePatch/issues/872 I made earlier is to put a delay on the demolition ability (of any spawning unit).

**Comment by ExiLe-ZH** [comment]
Damage is perhaps a factor to look into though, how does it compare to a regular terrorist? And does it differ per unit?

**Comment by commy2** [comment]
> And does it differ per unit?

No, exception is the Terror Bike, as long as it carries a Terrorist.

> Another suggestion in a topic https://github.com/TheSuperHackers/GeneralsGamePatch/issues/872 I made earlier is to put a delay on the demolition ability (of any spawning unit).

This would require engine changes unfortunately, probably

**Comment by MTKing4** [comment]
- nerf Tox to 4/6/10 Tox Rebels (from 4/8/16)
- nerf Demo to 3/5/8 Rebels (from 4/6/10)

**Comment by ExiLe-ZH** [comment]
> * nerf Tox to 4/6/10 Tox Rebels (from 4/8/16)
> 
>     * nerf Demo to 3/5/8 Rebels (from 4/6/10)

Sounds pretty good.

**Comment by xezon** [comment]
I do not like this. Before nerfing anything with Rebel Ambush it would make more sense to nerf Demo Bikes and Toxin RPG because these are the reason for overpoweredness.

**Comment by ExiLe-ZH** [comment]
Do we have issue reports for those? 

I think Gamma Busses are fun and not always easy to micro perfectly, perhaps a small nerf is justified, but just removing stinger tracking and the double fire bug (air + ground) might be enough already. That said, Busses in general will become a whole lot stronger when the Toxin/Radiation bunker bug is fixed. 

As for Demo Bikes, just introduce friendly fire (only vs other DB's) so that you can't brainlessly one-click unstoppable massed up blobs at the enemy anymore. Killing one will chain explode the rest, the player will have space them out and plan his attacks more carefully.

**Comment by xezon** [comment]
Demo Bikes
* #138
* #604

Toxin Gamma
* #806

**Comment by commy2** [comment]
> nerf Tox to 4/6/10 Tox Rebels (from 4/8/16)

Why is the progression like that? First star gives you 4, second star only 2, third star 4 again?

Edit: lol I pressed edit on my phone instead of reply xD

**Comment by MTKing4** [comment]
> I do not like this. Before nerfing anything with Rebel Ambush it would make more sense to nerf Demo Bikes and Toxin RPG because these are the reason for overpoweredness.

I don't think so, those are two separate issues you're talking about, we're talking about the overpoweredness of the ambush itself, and it's pretty overpowered alright, one click army deleter, whatever you do with demo bikes have no effect on this whatsoever.

> Why is the progression like that? First star gives you 4, second star only 2, third star 4 again?

I don't think i can use increments to get to a balanced number of units, balance takes precedence over round numbers afterall


Imo 3/6/12 can be accepted (12 is a lot imo) but 2/4/8 is too little, but we need a lower value for demo that is bigger than 2/4/8

**Comment by ExiLe-ZH** [comment]
> > nerf Tox to 4/6/10 Tox Rebels (from 4/8/16)
> 
> Why is the progression like that? First star gives you 4, second star only 2, third star 4 again?

4/6/9 has a sensible mathematical progression (+50%) if so desired.

Imo +100% progression isn't strictly necessary, balance > mathematical sense.

**Comment by commy2** [comment]
> 4/6/9 has a sensible mathematical progression (+50%) if so desired.

It makes the second genpoint weak. It would only do half as much as the first.

> we need a lower value for demo that is bigger than 2/4/8

Why? It's still spawning basically 2 vanilla Terrorists for free everywhere on the map, on top of an army possibly. Severely cutting down the numbers may mean the GLA has to at least spent a few points to get a reliable boom that razes buildings.

> balance > mathematical sense

They're both the same thing.

This would benefit from some stats which buildings you can blow up at each level.

**Comment by ExiLe-ZH** [comment]
> > 4/6/9 has a sensible mathematical progression (+50%) if so desired.
> 
> It makes the second genpoint weak. It would only do half as much as the first.
> 

You also suggested 6 with level 2, I think it's a well balanced number. 8 from the original is already very deadly vs Vees and Buggies and people often combine it with level 1 Bounty.

But I like 3 instead of 4 for level 1 aswell, like you suggested. 3 Rebels are still useful for clearing buildings and capturing vehicles e.g. and level 2 will relatively become a better upgrade.

12 is still a nerf over the original, but I'm not sure if it's significant enough. 9 or 10 sounds better to me in terms of balance/deadliness, but ofcourse it needs to be attractive over level 2, otherwise people will just skip it and always get Bounty.

**Comment by xezon** [comment]
The proper way to mitigate this is have a Demo Ability cooldown for Ambush Rebels.

**Comment by ExiLe-ZH** [comment]
> The proper way to mitigate this is have a Demo Ability cooldown for Ambush Rebels.

Think commy said it's technically hard or impossible to implement? 

But if possible it would be great, the cooldown has to be pretty fast though, because you should be able to pull off some damage. Otherwise it will become pretty useless compared to Toxin Rebel Ambush, which screws Tox vs Demo, but also nerfs Demo vs other factions.

I also wonder how much damage 9 instead 10 Demo Rebels do versus the new IC?

**Comment by xezon** [comment]
In worst case requires feature with Thyme to enable this functionality.

**Comment by xezon** [comment]
Jundiyy

> It's not a must but feel like GLA Ambush could do with a slight buff, perhaps an extra Rebel. It's quite weak and usually used to take a sniped vehicle or as a distraction. Not much else you can do and GLA doesn't have any other powerful GPs apart from rank 5.

> Another way to nerf Demo Ambush is to increase the timer.
Reducing the amount of Rebels does work but usually 2 or 3 are enough to destroy an army, so it doesn't exactly solve the problem. Delaying the Ambush by 30 to 60 seconds means the reset happens later every time, making it weaker.

> For Tox, reducing Rebels or increasing timer would be enough.

**Comment by commy2** [comment]
I added a 3 second initial wait time for Suicide ability in this pull https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2303.

Demonstration:

https://github.com/TheSuperHackers/GeneralsGamePatch/assets/6576312/1eb1d336-8de9-4b42-9528-09c398e516a9

(in video it was 5 seconds)

This initial delay is configurable. What is not possible is to give different units different delays, but I do not think that is necessary.

**Comment by xezon** [comment]
So how long should this delay be then?

**Comment by xezon** [comment]
3 seconds are probably too long. By that time all Rebels could be shot dead already. I assume it needs to be somewhere around 1 to 2 seconds.

**Comment by commy2** [comment]
It should be enough time, so a blob of units can scatter with X and avoid total destruction.

**Comment by xezon** [comment]
It is not clear to me if that is ok. There can also be Gattling or Dragon Tanks that kill the Terrorists fast.

**Comment by MTKing4** [comment]
3s is a good start for now, I'll add it to my test list to test this with @ImTimK hopefully soon

**Comment by ExiLe-ZH** [comment]
Oh cool, thought this wasn't possible without Thyme. I'm up to test this.

3 seconds sounds very long though, demolition ability shouldn't become completely useless versus units. 1-2 seconds sounds more appropriate?

The Rebels are controllable right? So you can press X to follow the opponent's units before detonating?

Overall I'm not sure what's needed balance wise, Demo is already in a pretty good spot, I believe it's close 50% winrate overall.

**Comment by commy2** [comment]
> The Rebels are controllable right? 

Yes, it's just that the Suicide ability starts on a special ability cooldown. Think of it like Helix Bomb, Booby Traps or Capture Buildings abilities.

**Comment by ExiLe-ZH** [comment]
> Overall I'm not sure what's needed balance wise, Demo is already in a pretty good spot, I believe it's close 50% winrate overall. 

One thing it makes worse is demo vs tox (and other gla's), it brings it further in favor of tox. 

So Tox probably needs a rebel reduction to balance it out, which also might improve other matchups since Tox is positive vs most armies.
